src/transformers/generation/beam_search.py

```diff
@@ -956,12 +954,16 @@ def add(
         hyp: torch.LongTensor,
         sum_logprobs: float,
         beam_indices: Optional[torch.LongTensor] = None,
-        decoder_prompt_len: Optional[int] = 0,
+        generated_len: Optional[int] = None,
     ):
         """
         Add a new hypothesis to the list.
         """
-        score = sum_logprobs / ((hyp.shape[-1] - decoder_prompt_len) ** self.length_penalty)
+        if generated_len is not None:
```

Comment:
I'd add a note that the `else` case here exists for retrocompatibility reasons :)